lib/src/models/parsed_credential.dart

```diff
@@ -0,0 +1,31 @@
+import 'package:freezed_annotation/freezed_annotation.dart';
+import 'package:ledger_cardano/src/utils/validation_exception.dart';
+
+part 'parsed_credential.freezed.dart';
+
+@freezed
+class ParsedCredential with _$ParsedCredential {
+  ParsedCredential._();
+
+  factory ParsedCredential.keyPath({
+    required List<int> path,
+  }) = CredentialKeyPath;
+
+  factory ParsedCredential.keyHash({
+    required String keyHashHex,
+  }) {
+    if (keyHashHex.length != 56) {
+      throw ValidationException("Key hash hex must be exactly 56 characters long.");
+    }
+    return ParsedCredential.keyHash(keyHashHex: keyHashHex);
+  }
```

Comment:
This is an infinite loop. You can achieve this by either:
a) checking inside constructor if it's `CredentialKeyHash` (which will also smart cast) and then checking the keyHashHex length.
b) creating and using an extra private factory
```dart
factory ParsedCredential._keyHash({
    required String keyHashHex,
  }) = CredentialKeyHash;
```

and changing the `return` of `ParsedCredential.keyHash` to use the new private factory: `ParsedCredential._keyHash`.